libvast/src/detail/posix.cpp

```diff
@@ -226,33 +223,41 @@ bool poll(int fd, int usec) {
   if (rc < 0) {
     switch (rc) {
       default:
-        VAST_RAISE_ERROR(std::logic_error, "unhandled select() error");
+        VAST_RAISE_ERROR(std::logic_error, "unhandled select(2) error");
```

Comment:
Can we return an error here as well now?

Reply:
I'd rather call `vast::die` here.